### PR TITLE
added create api key

### PR DESCRIPTION
**Pull request by Ragavi916**
Added create api Key function

### CODE REVIEW

File: client/api_key.go [hunk 1]
```diff
@@ -5,11 +5,46 @@ package client
...
+	log "github.com/sirupsen/logrus"
+	"time"
 
 	clientv1 "github.com/spectrocloud/palette-sdk-go/api/client/v1"
 	"github.com/spectrocloud/palette-sdk-go/api/models"
 )
 
+// CreateAPIKey creates a new API key for the specified tenant.
+func (h *V1Client) CreateAPIKey(name, TenantUID, emailID, org, password string) (string, error) {
+	annotations := map[string]string{
```
Comment (TylerGillson) [inline client/api_key.go:18]: This is a public repo. Please make annotations a param and remove the hardcoded QA annotations.

File: client/api_key.go [hunk 2]
```diff
@@ -5,11 +5,46 @@ package client
...
+		"Description": "Automation-Test"}
+	body := &models.V1APIKeyEntity{
+		Metadata: &models.V1ObjectMeta{
+			Name:        name,
+			Annotations: annotations,
+		},
+		Spec: &models.V1APIKeySpecEntity{
+			UserUID: TenantUID,
+			Expiry:  models.V1Time(time.Now().Add(5 * time.Hour)), // Set expiry time
```
Comment (TylerGillson) [inline client/api_key.go:27]: Please make expiry a param.

File: client/api_key.go [hunk 3]
```diff
@@ -5,11 +5,46 @@ package client
...
+	"github.com/go-openapi/strfmt"
+	log "github.com/sirupsen/logrus"
+	"time"
 
 	clientv1 "github.com/spectrocloud/palette-sdk-go/api/client/v1"
 	"github.com/spectrocloud/palette-sdk-go/api/models"
 )
 
+// CreateAPIKey creates a new API key for the specified tenant.
+func (h *V1Client) CreateAPIKey(name, TenantUID, emailID, org, password string) (string, error) {
```
Comment (TylerGillson) [inline client/api_key.go:17]: Please remove`TenantUID`, `emailID`, `org`, and `password`. You can fetch the tenantUID using `GetTenantUID` from `tenant.go`.

File: client/api_key.go [hunk 4]
```diff
@@ -5,11 +5,46 @@ package client
...
+		Metadata: &models.V1ObjectMeta{
+			Name:        name,
+			Annotations: annotations,
+		},
+		Spec: &models.V1APIKeySpecEntity{
+			UserUID: TenantUID,
+			Expiry:  models.V1Time(time.Now().Add(5 * time.Hour)), // Set expiry time
+		},
+	}
+	if _, err := h.AuthLogin(emailID, org, password); err != nil {
```
Comment (TylerGillson) [inline client/api_key.go:30]: Please remove L30-38. None of this authentication logic belongs here. The user should instantiate their Palette client however they want _before_ calling `CreateAPIKey`

File: client/api_key.go (hunk 1 above)
Comment (Ragavi916) [reply client/api_key.go:18]: > This is a public repo. Please make annotations a param and remove the hardcoded QA annotations.

Removed the hardcoded annotations

File: client/api_key.go [hunk 5]
```diff
@@ -44,3 +79,19 @@ func (h *V1Client) DeleteAPIKey(uid string) error {
 	}
 	return nil
 }
+
+// AuthLogin authenticates a user and returns a session token.
+func (h *V1Client) AuthLogin(emailID, org, password string) (string, error) {
```
[the remark this answers is not in the document]
Comment (Ragavi916) [reply client/api_key.go:84]: > An equivalent method already exists in `user.go`. Please remove.

yes removed it. For authentication purpose, I can use from user.go

File: client/api_key.go (hunk 2 above)
Comment (Ragavi916) [reply client/api_key.go:27]: > Please make expiry a param.

Done

File: client/api_key.go (hunk 4 above)
Comment (Ragavi916) [reply client/api_key.go:30]: yes removed and authenticated before calling CreateAPIKey

File: client/api_key.go (hunk 3 above)
Comment (Ragavi916) [reply client/api_key.go:17]: Done..Using GetTenantUID for fetching  the tenant id

**Review by TylerGillson — DISMISSED** [review]
Nice! Looks great 🚀 

You'll need to add a docstring comment to the new function for the linter to pass though.

**Comment by Ragavi916** [comment]
> Nice! Looks great 🚀
> 
> You'll need to add a docstring comment to the new function for the linter to pass though.

ok sure, thanks

**Review by TylerGillson — APPROVED** [review]

**Pull request merged**